kbase-extension/static/kbase/js/common/cellComponents/tabs/results/outputWidget.js

```diff
@@ -1,39 +1,79 @@
 /**
  * Should get fed data to view.
  */
-define(['bluebird', 'common/html', 'common/ui', 'util/kbaseApiUtil'], (
-    Promise,
-    html,
-    UI,
-    APIUtil
-) => {
+define([
+    'bluebird',
+    'base/js/namespace',
+    'uuid',
+    'common/html',
+    'common/ui',
+    'util/kbaseApiUtil',
+    'common/events',
+    'jquery',
+    'jquery-dataTables',
+], (Promise, Jupyter, UUID, html, UI, APIUtil, Events, $) => {
     'use strict';
 
     const tag = html.tag,
         div = tag('div'),
         tr = tag('tr'),
         th = tag('th'),
         td = tag('td'),
-        table = tag('table');
+        thead = tag('thead'),
+        tbody = tag('tbody'),
+        table = tag('table'),
+        a = tag('a'),
+        tablePageLength = 50;
 
     function OutputWidget() {
         let container, ui;
 
         /**
          *
-         * @param {Array} objectData an array of report object references
+         * @param {Array} objectData an array of object data. Each element is expected
+         *  to have these properties:
+         *  - type - the workspace object type
+         *  - name - the name of the object
+         *  - description - a description of the object
          */
         function renderOutput(objectData) {
             if (objectData.length === 0) {
-                return div('No objects created');
+                return {
+                    layout: div('No objects created'),
+                };
             }
-            return table({ class: 'table table-bordered table-striped' }, [
-                tr([th('Created Object Name'), th('Type'), th('Description')]),
-                ...objectData.map((obj) => {
-                    const parsedType = APIUtil.parseWorkspaceType(obj.type);
-                    return tr([td(obj.name), td(parsedType.type), td(obj.description)]);
-                }),
-            ]);
+            const events = Events.make();
+            const layout = table(
+                {
+                    class: 'table table-bordered',
+                    dataElement: 'objects-table',
+                },
+                [
+                    thead(tr([th('Created Object Name'), th('Type'), th('Description')])),
+                    tbody([
+                        ...objectData.map((obj) => {
+                            const parsedType = APIUtil.parseWorkspaceType(obj.type);
+                            const objLink = a(
+                                {
+                                    href: '#',
+                                    style: 'cursor: pointer;',
```

Comment:
can you give this link a class and then move this style declaration into the scss file?

Reply:
Yup, done.